### ISSUE TITLE
Question about the evaluation_gpu.py

### ISSUE THREAD

**Issue by viet2411** [layumi/Person_reID_baseline_pytorch#344]
Hello,

First of all, thank you for your work.
I'm working on two vehicle dataset using your code, but the images from test set does not have camera ID, so the program returned 0 for all results.
To overcome this problem, should I randomly put camera ID for each image, or should I change the junk_index in the evaluation_gpu.py to empty array [ ] ?

I'm looking forward to hearing your answer.
Thank you and best regards.

**Comment by layumi** [comment]
Hi @viet2411 

Object reid assumes **Cross-Camera** Retrieval. 

Therefore, we set to ignore the same-camera return result even if it is correct. 

1. If you want to retrieval same-camera results.  
You only need to ignore this line to enable it  https://github.com/layumi/Person_reID_baseline_pytorch/blob/master/evaluate_gpu.py#L25  


2. If you have images from different but unknown cameras. 
Yes. You can simply set the gallery image camera as {2,3,4,5,6,7,8,.....N}  as different cameras.

**Comment by viet2411** [comment]
Thank you for your quick response. I have one more question about the training process.

I want to use only the Triplet loss, so I comment out the line 'loss = criterion(logits, labels)' in train.py, so the program will only calculate triplet loss and not CE loss.
But this resulted in bad training accuracy (Training accuracy did not get over 1%)
Did I do something wrong here? Or did I misunderstand the program here?

I'm looking forward to hearing your answer.
Thank you and best regards.

**Comment by layumi** [comment]
If you only use triplet loss, the final classifier will not be trained. so the classification accuracy is random, and close to 1/751.

**Comment by viet2411** [comment]
Thank you for your answer. I understood the code now.
I tried to train the model on Market dataset with the above setting, using only the Triplet loss from pytorch_metric_learning library, and the test result was surprisingly bad:

torch.Size([3368, 512])
Rank@1:0.428741 Rank@5:0.666271 Rank@10:0.752078 mAP:0.270721

Have you tried this method before? What do you think about the result here?
If possible, I'd like to hear your opinion.

Thank you and best regards.

**Comment by layumi** [comment]
Hi @viet2411 

Yes. It is possible. I do not optimise the triplet loss from metric learning. 

You may refer to my hand-crafted triplet loss at https://github.com/layumi/Person-reID-triplet-loss , which achieves a reasonable performance.

**Comment by viet2411** [comment]
Thank you for your response!
